vars/pssh.groovy

```diff
@@ -2,12 +2,27 @@
 
 void call(Set<String> hosts, String command) {
   sh """
-    pssh \
-      -H "${hosts.sort().join(' ')}" \
-      -O StrictHostKeyChecking=no \
-      -O UserKnownHostsFile=/dev/null \
-      -t 300 \
-      --inline \
-      "${command}"
+    hosts="${hosts.sort().join(' ')}"
+    tmpdir=/tmp/pssh.\$\$
+    mkdir -p \$tmpdir
+
+    for host in \${hosts[@]}; do
+      timeout 300 ssh \
+        -o StrictHostKeyChecking=no \
+        -o UserKnownHostsFile=/dev/null \
+        -o ConnectTimeout=10 \
+        -o BatchMode=yes \
+        \${host} \
+        "${command}" > \${tmpdir}/\${host} 2>&1 &
+    done
+    
+    wait
```

Comment:
I think we'll need to adjust this so that this script returns the appropriate exit code if any of the commands fail. We've [used a wait loop before](https://github.com/mysidewalk/dev-ops/blob/master/gce/postgresql/create.sh#L66), but [SO has a pretty thorough set of examples](https://stackoverflow.com/questions/356100/how-to-wait-in-bash-for-several-subprocesses-to-finish-and-return-exit-code-0) if you want to dig into it.

Reply:
Yeah that's a good idea too. It now throws the latest background pid into an array and then waits and checks each exit code. If any are bad we will throw a message and exit after spitting the output from the commands in the job output.